volatility3/framework/layers/intel.py

```diff
@@ -102,70 +104,84 @@ def _translate(self, offset: int) -> Tuple[int, int, str]:
         translated address lives in and the layer_name that the address
         lives in
         """
-        entry, position = self._translate_entry(offset)
+        # Setup the entry and how far we are through the offset
+        # Position maintains the number of bits left to process
+        # We or with 0x1 to ensure our page_map_offset is always valid
+        position = self._initial_position
+        entry = self._initial_entry
+
+        if self.minimum_address > offset > self.maximum_address:
+            raise exceptions.PagedInvalidAddressException(self.name, offset, position + 1, entry,
+                                                          "Entry outside virtual address range: " + hex(entry))
+
+        entry, position = self._translate_entry(offset, position, entry)
 
         # Now we're done
         if not self._page_is_valid(entry):
-            raise exceptions.PagedInvalidAddressException(self.name, offset, position + 1, entry,
-                                                          f"Page Fault at entry {hex(entry)} in page entry")
+            return self._handle_page_fault(self.name, offset, position + 1, entry, f"Page Fault at entry {hex(entry)} in page entry")
+
         page = self._mask(entry, self._maxphyaddr - 1, position + 1) | self._mask(offset, position, 0)
 
         return page, 1 << (position + 1), self._base_layer
 
-    def _translate_entry(self, offset: int) -> Tuple[int, int]:
+    def _find_structure_index(self, position: int) -> Tuple[str, int, bool]:
+        if not self._structure_position_table:
+            counter = self._initial_position
+            for name, size, large_page in self._structure:
+                self._structure_position_table[counter] = name, size, large_page
+                counter -= size
+            # print(self._structure_position_table)
```

Comment:
It appears to be a code comment that has not been removed.

Reply:
Removed the commented print